ros2_foxglove_bridge/src/json_to_ros.cpp

```diff
@@ -0,0 +1,139 @@
+      break;
+    case MessageType::Bool:
+      field = value.get<bool>();
+      break;
+    case MessageType::Octet:
+    case MessageType::UInt8:
+      field = value.get<uint8_t>();
+      break;
+    case MessageType::Int8:
+      field = value.get<int8_t>();
+      break;
+    case MessageType::UInt16:
+      field = value.get<uint16_t>();
+      break;
+    case MessageType::Int16:
+      field = value.get<int16_t>();
+      break;
+    case MessageType::UInt32:
+      field = value.get<uint32_t>();
+      break;
+    case MessageType::Int32:
+      field = value.get<int32_t>();
+      break;
+    case MessageType::UInt64:
+      field = value.get<uint64_t>();
+      break;
+    case MessageType::Int64:
+      field = value.get<int64_t>();
+      break;
+    case MessageType::String:
+      field = value.get<std::string>();
+      break;
+    case MessageType::WString:
+      field = value.get<std::wstring>();
+      break;
+    case MessageType::Compound: {
+      // Recursively convert compound messages
+      auto& compound = field.as<ros2_babel_fish::CompoundMessage>();
+      for (const auto& key : compound.keys()) {
+        assignJsonToMessageField(compound[key], value[key]);
+      }
+      break;
+    }
+    case MessageType::Array: {
+      // Ensure the JSON value is an array
+      if (!value.is_array()) {
+        break;
+      }
+
+      auto& array = field.as<ros2_babel_fish::CompoundArrayMessage>();
+      if (array.isFixedSize() || array.isBounded()) {
+        const size_t limit = std::min(array.maxSize(), value.size());
+        for (size_t i = 0; i < limit; ++i) {
+          auto& arrayEntry = array.size() > i ? array[i] : array.appendEmpty();
+          assignJsonToMessageField(arrayEntry, value[i]);
+        }
+      } else {
+        array.clear();
+        for (const auto& jsonArrayEntry : value) {
+          auto& arrayEntry = array.appendEmpty();
+          assignJsonToMessageField(arrayEntry, jsonArrayEntry);
+        }
+      }
+      break;
+    }
+  }
+}
+
+std::optional<std::exception> jsonMessageToRos(
+  const std::string_view jsonMessage, const std::string& schemaName,
+  ros2_babel_fish::BabelFish::SharedPtr babelFish,
+  ros2_babel_fish::CompoundMessage::SharedPtr& outputMessage) {
+  // Decode the JSON message
+  nlohmann::json json;
+  try {
+    json = nlohmann::json::parse(jsonMessage);
+  } catch (const nlohmann::json::parse_error& e) {
+    return e;
+  }
+
+  // Convert the JSON message to a ROS message using ros2_babel_fish
+  ros2_babel_fish::CompoundMessage::SharedPtr rosMsgPtr;
+  try {
+    rosMsgPtr = babelFish->create_message_shared(schemaName);
+  } catch (const ros2_babel_fish::BabelFishException& e) {
+    return e;
+  }
+  auto& rosMsg = *rosMsgPtr;
+  for (const auto& key : rosMsg.keys()) {
+    if (!json.contains(key)) {
+      continue;
+    }
+
+    try {
+      assignJsonToMessageField(rosMsg[key], json[key]);
+    } catch (const std::exception& e) {
+      return e;
+    }
+  }
```

Comment:
that's a lot of try-catch blocks. Having only one try-catch block should make it more readable

Reply:
Or related to my previous comment: Don't catch exceptions here and return the output message rather than the std::optional